### DISCUSSION TITLE
mvn install failed for OpenAM RIA Web UI

### DISCUSSION THREAD

**Discussion by asis2016** [OpenIdentityPlatform/OpenAM#608] [General]
Hi all,
When I tried to mvn install -f OpenAM, it fails for OpenAM RIA Web UI.

---
[INFO] --- frontend-maven-plugin:1.7.6:npm (npm-install) @ openam-ui-ria ---
[INFO] Running 'npm install' in /root/OpenAM/openam-ui/openam-ui-ria
[INFO] 
[INFO] > phantomjs-prebuilt@2.1.13 install /root/OpenAM/openam-ui/openam-ui-ria/node_modules/phantomjs-prebuilt
[INFO] > node install.js
[INFO] 
[ERROR] sh: 1: node: Permission denied
[WARNING] npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
[WARNING] npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
[ERROR] 
[ERROR] npm ERR! code ELIFECYCLE
[ERROR] npm ERR! syscall spawn
[ERROR] npm ERR! file sh
[ERROR] npm ERR! errno ENOENT
[ERROR] npm ERR! phantomjs-prebuilt@2.1.13 install: `node install.js`
[ERROR] npm ERR! spawn ENOENT
[ERROR] npm ERR! 
[ERROR] npm ERR! Failed at the phantomjs-prebuilt@2.1.13 install script.
[ERROR] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[ERROR] 
[ERROR] npm ERR! A complete log of this run can be found in:
[ERROR] npm ERR!     /root/.npm/_logs/2023-04-22T07_52_16_479Z-debug.log
---

**Comment by maximthomas** [comment]
@asis2016, could you provide more details about your build environment and also attach /root/.npm/_logs/2023-04-22T07_52_16_479Z-debug.log?

**Reply by asis2016** [reply]
Thanks for the reply.
Here is my detail:
openjdk 11.0.18
Apache Maven 3.6.3
Apache Ant(TM) version 1.10.12

And log file
[2023-04-24T18_59_26_795Z-debug.log](https://github.com/OpenIdentityPlatform/OpenAM/files/11313786/2023-04-24T18_59_26_795Z-debug.log)

**Reply by maximthomas** [reply]
Could not repouce the issue. I took maven docker image https://hub.docker.com/_/maven `maven:3-eclipse-temurin-11`, run it in console
```bash
docker run -it --rm --name build-openam maven:3-eclipse-temurin-11 /bin/bash
```
and executed the following commands in the Docker container terminal

```bash
export OPENSSL_CONF=/dev/null
apt update
apt install bzip2
git clone https://github.com/OpenIdentityPlatform/OpenAM.git
mvn install -f OpenAM
```
The build completed successfully

**Reply by asis2016** [reply]
Thanks it works.
